pxr/usd/sdf/testenv/testSdfReference.py

```diff
@@ -27,6 +27,84 @@
 from pxr import Sdf, Tf
 import unittest
 
```

Comment:
I'm glad to see test cases added for SdfAssetPath, but I don't feel they belong in testSdfReference (it's not obvious to look for them here.) Can we instead add a new testSdfAssetPath to hold these tests? I'm a little surprised we don't have a testSdfAssetPath already.
Additionally, we typically make one unittest.TestCase class and have multiple test case methods for each case. That would make more sense here too for the trio of SdfAssetPath tests.